### PR TITLE
FlowRunTask improvements

### LINKED ISSUE #2997: Expose idempotency key in `FlowRunTask`
We should:
- include `map_index` in the idempotency key of `FlowRunTask` by default
- allow users to override this behavior and provide a custom key at runtime (with the necessary caveats)

### LINKED ISSUE #3001: Missing key 'idempotency_key' in Server-mode GraphQL schema (found in v0.12.4)
## Description
When starting a registered flow from another flow using `prefect.tasks.prefect.flow_run.FlowRunTask` an error occurred: 
`requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://localhost:4200/graphql`.  At the same time the Apollo log states

```
2020-07-20T13:21:31.790Z {"message":"Variable \"$input\" got invalid value { flow_id: \"80de1b39-fed7-47ab-a5a5-fc22ac6f87d2\", idempotency_key: \"6a83567d-7e10-4b3a-815c-a29fe3ecab02\" }; Field \"idempotency_key\" is not defined by type create_flow_run_input.","locations":[{"line":1,"column":10}],"extensions":{"code":"INTERNAL_SERVER_ERROR"}}
```

The problem exists in [schema.graphql line 114](https://github.com/PrefectHQ/prefect/blob/0.12.4/server/src/prefect_server/graphql/schema/schema.graphql#L114) . There is *no* `idempotency_key` key in the `create_flow_input` structure/input. The [Client Implementation](https://github.com/PrefectHQ/prefect/blob/0.12.4/src/prefect/client/client.py#L869) is not forcing the usage of `idempotency.py` because its optional. But the `FlowRunTask` is setting a default in [flow_run.py:122](https://github.com/PrefectHQ/prefect/blob/0.12.4/src/prefect/tasks/prefect/flow_run.py#L122). 

You can also find the key missing by using the (standalone/server-mode) UI using the interactive API under _DOCS_ -> _Mutations_ -> `create_flow` ->  _Arguments_ -> `input::CREATE_FLOW_INPUT`.

When this key is present, the start of a flow run must fail because of the unexpected key. While hunting down this bug I used my own `FlowRunTask` wich was an exact copy of Prefect's FlowRunTask, *except line 122 removed*. As a result I could initiate a flow run like expected. 

Maybe its sufficient to add `idempotency_key` to the _schema.graphql_ file?

## Expected Behavior
The registered flow should start. Usage of `idempotency_key` should be possible.

## Reproduction
Minimal example to reproduce:

### flow1.py
```python
#!/usr/bin/env python
# coding: utf-8

import prefect
from prefect import task, Flow
from prefect.environments.storage.local import Local


@task
def task1():
    prefect.context.get("logger").info("hello, task1")


with Flow("flow1", storage=Local(directory="/flows/.prefect/flows")) as flow:
    task1()

if __name__ == "__main__":
    flow.register()
```

### flow2.py

```python
#!/usr/bin/env python
# coding: utf-8

import prefect
from prefect import task, Flow
from prefect.tasks.prefect import FlowRunTask
from prefect.environments.storage.local import Local


@task
def task2():
    prefect.context.get("logger").info("hello, task2")


with Flow("flow2", storage=Local(directory="/flows/.prefect/flows")) as flow:
    # the task will generate an idempotency_key if not present (or None!)
    flow1_run = FlowRunTask(name="flow1-runner", flow_name="flow1")
    task2.set_downstream(flow1_run)

if __name__ == "__main__":
    flow.register()
```

### Test run (fail)
In the Agent's container in the path where I store my flows:
```
prefect  run server -n flow2 -w
```

## Environment
Server mode; Prefect Server and all other components are at v0.12.4.

### PR DESCRIPTION
**Pull request by cicdw**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] add a changelog entry in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
This PR makes various improvements to the increasingly popular `FlowRunTask`:
- sensible naming to reflect the underlying Flow
- user-provided idempotency keys
- taking `map_index` into account for default idempotency keys
- not using `idempotency_key` for server runs
- adds a Flow-to-Flow idiom
- exposing `context` configuration for the newly created flow run

## Why is this PR important?
Closes #2997 
Closes #3001 
Closes https://github.com/PrefectHQ/prefect/issues/2941

### CODE REVIEW

File: src/prefect/tasks/prefect/flow_run.py [hunk 1]
```diff
@@ -114,12 +123,19 @@ def run(
         # grab the ID for the most recent version
         flow_id = flow[0].id
 
+        if is_hosted_backend and context.get("flow_run_id"):
```
Comment (jlowin) [inline src/prefect/tasks/prefect/flow_run.py:126]: The upcoming server release will support idempotency keys so I'm going to open an issue to revert this trap

**Comment by lauralorenz** [comment]
@cicdw can I sneak https://github.com/PrefectHQ/prefect/issues/2941 in here really quick?

File: src/prefect/tasks/prefect/flow_run.py (hunk 1 above)
Comment (cicdw) [reply src/prefect/tasks/prefect/flow_run.py:131]: I can add a TODO also

File: docs/core/idioms/flow-to-flow.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+Oftentimes different people are responsible for maintaining different flows; in this case it can be useful
+to construct a Flow-of-Flows that specifies stateful dependencies between various Flows.  The `wait` kwarg
+allows you to specify that the task should wait until the triggered flow run completes, and reflect the
+flow run state as the task state.
+
+The following example creates the following Flow-of-Flows that runs every weekday:
+
+![Flow of Flows](/idioms/flow-of-flows.png)
+
+:::: tabs
```
[the remark this answers is not in the document]
Comment (cicdw) [reply docs/core/idioms/flow-to-flow.md:75]: @joshmeek I couldn't reproduce locally - let's see if it happens again

[the remark this answers is not in the document]
Comment (cicdw) [reply docs/core/idioms/flow-to-flow.md:75]: Looks like it's formatted properly 👍

**Pull request merged**